frontend/src/wrapper/TopNavigation.tsx

```diff
@@ -0,0 +1,186 @@
+import { Add, Settings } from '@mui/icons-material'
+import DarkModeIcon from '@mui/icons-material/DarkMode'
+import LogoutIcon from '@mui/icons-material/Logout'
+import MenuIcon from '@mui/icons-material/Menu'
+import {
+  Box,
+  IconButton,
+  ListItemIcon,
+  ListItemText,
+  Menu,
+  MenuItem,
+  MenuList,
+  Stack,
+  Switch,
+  Toolbar,
+  Typography,
+} from '@mui/material'
+import MuiAppBar, { AppBarProps as MuiAppBarProps } from '@mui/material/AppBar'
+import { styled, useTheme } from '@mui/material/styles'
+import Image from 'next/legacy/image'
+import { useRouter } from 'next/router'
+import { MouseEvent, useContext, useState } from 'react'
+
+import { EntityKind, User } from '../../types/types'
+import ExpandableButton from '../common/ExpandableButton'
+import UserAvatar from '../common/UserAvatar'
+import ThemeModeContext from '../contexts/themeModeContext'
+import Link from '../Link'
+
+const drawerWidth = 240
+
+type TopNavigationProps = {
+  drawerOpen?: boolean
+  toggleDrawer: () => void
+  pageTopStyling?: any
```

Comment:
I think you can use the CSSProperties type provided by react: `import { CSSProperties } from 'react'